tests/backup/backup_helper.go

```diff
@@ -1500,7 +1504,85 @@ func IsPresent(dataSlice interface{}, data interface{}) bool {
 		}
 	}
 	return false
+}
+
+// getStorkImageVersion returns current stork image version.
+func getStorkImageVersion() (string, error) {
+	storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return "", err
+	}
+	storkImage := storkDeployment.Spec.Template.Spec.Containers[0].Image
+	storkImageVersion := strings.Split(storkImage, ":")[len(strings.Split(storkImage, ":"))-1]
+	return storkImageVersion, nil
+}
 
+// upgradeStorkVersion upgrades the stork to the provided version.
+func upgradeStorkVersion(storkImageToUpgrade string) error {
+	currentStorkImageStr, err := getStorkImageVersion()
+	if err != nil {
+		return err
+	}
+	currentStorkVersion, err := version.NewSemver(currentStorkImageStr)
+	if err != nil {
+		return err
+	}
+
+	storkImageVersionToUpgradeStr := strings.Split(storkImageToUpgrade, ":")[len(strings.Split(storkImageToUpgrade, ":"))-1]
+	storkImageVersionToUpgrade, err := version.NewSemver(storkImageVersionToUpgradeStr)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Current stork version : %s", currentStorkVersion)
+	log.Infof("Upgrading stork version to : %s", storkImageVersionToUpgrade)
+
+	if currentStorkVersion.GreaterThanOrEqual(storkImageVersionToUpgrade) {
+		log.Errorf("Cannot upgrade stork version from %s to %s as the current version is higher than the provided version", currentStorkVersion, storkImageVersionToUpgrade)
+		return nil
+	}
+
+	isOpBased, _ := Inst().V.IsOperatorBasedInstall()
```

Comment:
If Inst().V.IsOperatorBasedInstall throws an error, how about we handle it?

Reply:
we should not handle the error here, as IsOperatorBasedInstall() will return an error if it is non-operator based install, we just want the bool value to got forward with our upgrading.

Reply:
How about indicating in the log that the installation is not based on an operator?

Reply:
I've added the logs under if conditions